src/unix/linux-core.c

```diff
@@ -278,33 +281,46 @@ int uv_cpu_info(uv_cpu_info_t** cpu_infos, int* count) {
   if (statfile_fp == NULL)
     return UV__ERR(errno);
 
-  err = uv__cpu_num(statfile_fp, &numcpus);
-  if (err < 0)
-    goto out;
+  for (i = 0; i < max_retry; i += 1) {
+    err = uv__cpu_num(statfile_fp, &numcpus);
+    if (err < 0)
+      goto out;
 
-  err = UV_ENOMEM;
-  ci = uv__calloc(numcpus, sizeof(*ci));
-  if (ci == NULL)
-    goto out;
+    err = UV_ENOMEM;
+    ci = uv__calloc(numcpus, sizeof(*ci));
+    if (ci == NULL)
+      goto out;
 
-  err = read_models(numcpus, ci);
-  if (err == 0)
-    err = read_times(statfile_fp, numcpus, ci);
+    err = read_models(numcpus, ci);
+    if (err == 0)
+      numread = read_times(statfile_fp, numcpus, ci);
 
-  if (err) {
-    uv_free_cpu_info(ci, numcpus);
-    goto out;
-  }
+    /** The cpu_num observed from uv__cpu_num(), read_models() and read_times()
+     * may be different for online/offline CPU race. Restart the procedure if
+     * that is the case.
+     */
+    if (numread != numcpus) {
```

Comment:
UB: `numread` is undefined when `err != 0`

Reply:
Fixed.